src/Microsoft.AspNetCore.Sockets.Common/TextMessageFormatter.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    internal static class TextMessageFormatter
+    {
+        private const byte FieldDelimiter = (byte)':';
+        private const byte MessageDelimiter = (byte)';';
+        private const byte TextTypeFlag = (byte)'T';
+        private const byte BinaryTypeFlag = (byte)'B';
+        private const byte CloseTypeFlag = (byte)'C';
+        private const byte ErrorTypeFlag = (byte)'E';
+
+        public static bool TryFormatMessage(Message message, Span<byte> buffer, out int bytesWritten)
+        {
+            // Write the length as a string
+            int written = 0;
+            if (!message.Payload.Buffer.Length.TryFormat(buffer, out int lengthLen, default(TextFormat), EncodingData.InvariantUtf8))
+            {
+                bytesWritten = 0;
+                return false;
+            }
+            written += lengthLen;
+            buffer = buffer.Slice(lengthLen);
+
+            // We need at least 4 more characters of space (':', type flag, ':', and eventually the terminating ';')
+            // We'll still need to double-check that we have space for the terminator after we write the payload,
+            // but this way we can exit early if the buffer is way too small.
+            if (buffer.Length < 4)
+            {
+                // Discard everything we've written. We don't need to clear it though, we just return 0 bytes written.
+                bytesWritten = 0;
+                return false;
+            }
+            buffer[0] = FieldDelimiter;
+            if (!TryFormatType(message.Type, buffer.Slice(1, 1), out _))
+            {
+                bytesWritten = 0;
+                return false;
+            }
+            buffer[2] = FieldDelimiter;
+            buffer = buffer.Slice(3);
+            written += 3;
+
+            // Payload
+            if (message.Type == MessageType.Binary)
```

Comment:
Yes, the full implementation of the binary format is coming in a separate PR later. This was a trivial thing to add since the prefix is almost the same as for the text format.